Added file with instructions for preparing validation datasets (#1179)
opencv/open_model_zoo: datasets.md

# Dataset Preparation Guide

> If you want to use prepared configs to run the Accuracy Checker tool and the Model quantizer, you need to organize folders with validation datasets in a certain way.  Instructions for preparing validation data are described in this document.

Reviewer: ```suggestion
If you want to use prepared configs to run the Accuracy Checker tool and the Model Quantizer, you need to organize folders with validation datasets in a certain way.  Instructions for preparing validation data are described in this document.
```

Author: Fixed

## [ImageNet](http://image-net.org)

### How download dataset

To download images from ImageNet, you need to have an account and agree to the Terms of Access. Follow the steps below:
1. Go to the [ImageNet](http://www.image-net.org/) homepage
2. If you have an account, click `Login`. Otherwise, click `Signup` in the right upper corner, provide your data, and wait for a confirmation email
3. Log in after receiving the confirmation email and go to the `Download` tab
4. Select `Download Original Images`
5. You will be redirected to the Terms of Access page. If you agree to the Terms, continue by clicking Agree and Sign

> 6. Click one of the links in the Download as one tar file section to select it

Reviewer: ```suggestion
6. Click one of the links in the "Download as one tar file" section to select it
```

Author: Done

7. Unpack archive

To download annotation files, you need to follow the steps below:
* `val.txt`

>   1. Download arhive <http://dl.caffe.berkeleyvision.org/caffe_ilsvrc12.tar.gz>

Reviewer: ```suggestion
  1. Download [archive](http://dl.caffe.berkeleyvision.org/caffe_ilsvrc12.tar.gz)
```

Author: Done

>   2. Unpack archive

Reviewer: Shouldn't you only need to unpack one specific file?

Author: Fixed

* `val15.txt`
  1. Download annotation file <https://raw.githubusercontent.com/hujie-frank/SENet/master/ILSVRC2017_val.txt>
  2. Rename `ILSVRC2017_val.txt` to `val15.txt`

### Structure of directory with dataset

This section describes what structure of directory dataset must have to run the Accuracy Checker tool.

> * `<DATASET_DIR>` - root directory

Reviewer: I think these sections kinda give off the impression that each dataset must be placed in its own root directory, when that's not the case. I would suggest you do the following:

1. Define `<DATASET_DIR>` in the intro section.
2. Transform these sections into something like this:

```markdown
### File layout

To use this dataset with OMZ tools, make sure `<DATASET_DIR>` contains the following:

* `ILSVRC2012_img_val` - directory containing the ILSVRC 2012 validation images
* `val.txt` - annotation file used for ILSVRC 2012
* `val15.txt` - annotation file used for ILSVRC 2015

Author: Done

    * `ILSVRC2012_img_val` - directory containing the ILSVRC 2012 validation images
    * `val.txt` - annotation file used for ILSVRC 2012
    * `val15.txt` - annotation file used for ILSVRC 2015

### Datasets in dataset_definitions.yml

> * `imagenet_1000_classes` used for evaluation models trained on ILSVRC 2012 dataset with 1000 classes. (model examples: alexnet, vgg16)

Reviewer 2: probably it makes sense to add links on model descriptions or highlight model names

Author: Done

> * `imagenet_1000_classes` used for evaluation models trained on ILSVRC 2012 dataset for 1000 classes. (model examples: alexnet, colorization-v2)

Reviewer 2: ```suggestion
* `imagenet_1000_classes` used for evaluation models trained on ILSVRC 2012 dataset with 1000 classes. (model examples: alexnet, colorization-v2)
```

Author: Done

> * `imagenet_1000_classes` used for evaluation models trained on ILSVRC 2012 dataset for 1000 classes. (model examples: alexnet, colorization-v2)

Reviewer 2: colorization not trained on this dataset, it uses another model for verification its accuracy
```suggestion
* `imagenet_1000_classes` used for evaluation models trained on ILSVRC 2012 dataset for 1000 classes. (model examples: alexnet,  vgg16)
```

Author: Done

> * `imagenet_1000_classes` used for evaluation models trained on ILSVRC 2012 dataset with 1000 classes. (model examples: [alexnet](models/public/alexnet/alexnet.md), [vgg16](models/public/vgg16/vgg16.md))

Reviewer: ```suggestion
* `imagenet_1000_classes` used for evaluation models trained on ILSVRC 2012 dataset with 1000 classes. (model examples: [`alexnet`](models/public/alexnet/alexnet.md), [`vgg16`](models/public/vgg16/vgg16.md))
```

And so on throughout the document.

Author: Done

* `imagenet_1000_classes_2015` used for evaluation models trained on ILSVRC 2015 dataset with 1000 classes. (model examples: [se-resnet-152](models/public/se-resnet-152/se-resnet-152.md), [se-resnext-50](models/public/se-resnext-50/se-resnext-50.md))
* `imagenet_1001_classes` used for evaluation models trained on ILSVRC 2012 dataset with 1001 classes (background label + original labels). (model examples: [googlenet-v2-tf](models/public/googlenet-v2-tf/googlenet-v2-tf.md), [resnet-50-tf](models/public/resnet-50-tf/resnet-50-tf.md))

## [Common Objects in Context (COCO)](http://cocodataset.org/#home)

### How download dataset

To download COCO dataset, you need to follow the steps below:
1. Go to the [COCO](http://cocodataset.org/#home) website

> 2. Go to the `Dataset-> Download` tab

Reviewer: Seems like these steps are pretty pointless, since the next step just gives you the links to download. There already is a link to the dataset home page in the title of the section, so there seems to be no reason to link to it again.

Author: These steps have been deleted

3. Select [`2017 Val images`](http://images.cocodataset.org/zips/val2017.zip) and [`2017 Train/Val annotations`](http://images.cocodataset.org/annotations/annotations_trainval2017.zip) and download these archives
4. Unpack archives

### Structure of directory with dataset

This section describes what structure of directory dataset must have to run the Accuracy Checker tool.
* `<DATASET_DIR>` - root directory
    * `val2017` - directory containing the COCO 2017 validation images
    * `instances_val2017.json` - annotation file which used for object detection and instance segmentation tasks
    * `person_keypoints_val2017.json` - annotation file which used for human pose estimation tasks

### Datasets in dataset_definitions.yml

> * `ms_coco_mask_rcnn` used for evaluation models trained on COCO dataset for object detection and instance segmentation tasks. Background label + label map with 80 public available object categories are used. Annotations are saved in image id ascend order.

Reviewer: ```suggestion
* `ms_coco_mask_rcnn` used for evaluation models trained on COCO dataset for object detection and instance segmentation tasks. Background label + label map with 80 public available object categories are used. Annotations are saved in order of ascending image ID.
```

Author: Done

> * `ms_coco_detection_91_classes` used for evaluation models trained on COCO dataset for object detection tasks. Background label + original label map with 91 object categories are used (public available 80 categories). Annotations are saved in image id ascend order. (model examples: [faster_rcnn_resnet50_coco](models/public/faster_rcnn_resnet50_coco/faster_rcnn_resnet50_coco.md), [ssd_resnet50_v1_fpn_coco](models/public/ssd_resnet50_v1_fpn_coco/ssd_resnet50_v1_fpn_coco.md))

Reviewer: This seems incomplete/contradictory. Are there 91 categories or 80?

Author: Description has been updated, please re-check

* `ms_coco_detection_80_class_with_background` used for evaluation models trained on COCO dataset for object detection tasks. Background label + label map with 80 public available object categories are used. Annotations are saved in image id ascend order. (model examples: faster-rcnn-resnet101-coco-sparse-60-0001, ssd-resnet34-1200-onnx)

> * `ms_coco_detection_80_class_without_background` used for evaluation models trained on COCO dataset for object detection tasks. Label map with 80 public available object categories is used. Annotations are saved in image id ascend order. (model examples: ctdet_coco_dlav0_384, pelee-coco)

Reviewer 2: peleee ssd based model, it has background
```suggestion
* `ms_coco_detection_80_class_without_background` used for evaluation models trained on COCO dataset for object detection tasks. Label map with 80 public available object categories is used. Annotations are saved in image id ascend order. (model examples: ctdet_coco_dlav0_384, yolo-v3-tf)
```

Author: Done.

> * `ms_coco_keypoints` used for evaluation models trained on COCO dataset for human pose estimation tasks. Each annotation stores multiple keypoints for one image. (model examples: human-pose-estimation-0001, openpose-pose-caffe)

Reviewer 2: there is no openpose-pose-caffe model in OMZ
```suggestion
* `ms_coco_keypoints` used for evaluation models trained on COCO dataset for human pose estimation tasks. Each annotation stores multiple keypoints for one image. (model examples: human-pose-estimation-0001)
```

Author: It is my mistake. Model has been removed

* `ms_coco_single_keypoints` used for evaluation models trained on COCO dataset for human pose estimation tasks. Each annotation stores single keypoints for image, so several annotation can be associated to one image. (model examples: [single-human-pose-estimation-0001](models/public/single-human-pose-estimation-0001/description/single-human-pose-estimation-0001.md))

> ## [WIDER Face](http://shuoyang1213.me/WIDERFACE/)

Reviewer: ```suggestion
## [WIDER FACE](http://shuoyang1213.me/WIDERFACE/)
```

Author: Done
